src/pygambit/game.pxi

```diff
@@ -1100,47 +1100,83 @@ class Game:
             f"{funcname}(): {argname} must be Action or str, not {action.__class__.__name__}"
         )
 
-    def append_move(self, node: typing.Union[Node, str],
+    def append_move(self, nodes: typing.Union[NodeReference, NodeReferenceSet],
                     player: typing.Union[Player, str],
                     actions: typing.List[str]) -> None:
-        """Add a move for `player` at the terminal node `node`.  `node` becomes part of
+        """Add a move for `player` at terminal `nodes`.  All elements of `nodes` become part of
         a new information set, with actions labeled according to `actions`.
 
         Raises
         ------
         UndefinedOperationError
-            If `node` is not a terminal node, or `actions` is not a positive number.
+            If `nodes` are not all terminal, or `actions` is not a positive number.
         MismatchError
-            If `node` is a `Node` from a different game, or `player` is a `Player` from a
-            different game.
+            If an element from `nodes` is a `Node` from a different game,
+            or `player` is a `Player` from a different game.
+        ValueError
+            If `nodes` has non unique elements, or the `nodes` is empty.
         """
-        resolved_node = cython.cast(Node, self._resolve_node(node, "append_move"))
         resolved_player = cython.cast(Player, self._resolve_player(player, "append_move"))
-        if len(resolved_node.children) > 0:
-            raise UndefinedOperationError("append_move(): `node` must be a terminal node")
         if len(actions) == 0:
             raise UndefinedOperationError("append_move(): `actions` must be a nonempty list")
+
+        resolved_nodes = (
```

Comment:
The outer parentheses are redundant.  You can just use the square brackets for the list directly.